### ISSUE TITLE
Cannot start cloudstack-agent if there is vm not managed by cloudstack on the host

### ISSUE THREAD

**Issue by weizhouapache** [apache/cloudstack#8040]
the agent.log shows
```
2023-10-03 12:18:23,692 ERROR [cloud.agent.AgentShell] (main:null) (logid:) Unable to start agent:
javax.naming.ConfigurationException: Error while creating Agent with class [com.cloud.hypervisor.kvm.resource.LibvirtComputingResource]. [Root exception is java.lang.NumberFormatException: null]
    at com.cloud.agent.AgentShell.launchAgentFromClassInfo(AgentShell.java:434)
    at com.cloud.agent.AgentShell.launchAgent(AgentShell.java:415)
    at com.cloud.agent.AgentShell.start(AgentShell.java:511)
    at com.cloud.agent.AgentShell.main(AgentShell.java:541)
Caused by: java.lang.NumberFormatException: null
    at java.base/java.lang.Integer.parseInt(Integer.java:614)
    at java.base/java.lang.Integer.parseInt(Integer.java:770)
    at com.cloud.hypervisor.kvm.resource.LibvirtDomainXMLParser.parseDomainXML(LibvirtDomainXMLParser.java:311)
    at com.cloud.hypervisor.kvm.resource.LibvirtComputingResource.setupMemoryBalloonStatsPeriod(LibvirtComputingResource.java:1301)
    at com.cloud.hypervisor.kvm.resource.LibvirtComputingResource.configure(LibvirtComputingResource.java:1244)
    at com.cloud.agent.Agent.<init>(Agent.java:190)
    at com.cloud.agent.AgentShell.launchNewAgent(AgentShell.java:452)
    at com.cloud.agent.AgentShell.launchAgentFromClassInfo(AgentShell.java:431)
    … 3 more
```

the issue is because there are 2 vms which are not created by cloudstack on the host. The xml of the vm does not have rate period and bytes in the rng secion. It should be

```
<rng model='virtio'>
<rate period='5000' bytes='4096' />
<backend model='random'>/dev/random</backend>
</rng>
 ```

Stopping the vms and edit the xml definition of vms should fix the issue.

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~

~~~

##### CLOUDSTACK VERSION

~~~

~~~

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~

~~~

##### ACTUAL RESULTS

~~~

~~~

**Comment by harikrishna-patnala** [comment]
@weizhouapache do we have to parse these VMs in first place ? or do we need to skip this section if does not exist for such VMs ?

**Comment by weizhouapache** [comment]
> LibvirtComputingResource

I think option 2 is better. 
we could skip the vms if the name does not start with i-/r-/s-/v- (not ideal but should work).

**Comment by DaanHoogland** [comment]
> > LibvirtComputingResource
> 
> I think option 2 is better. we could skip the vms if the name does not start with i-/r-/s-/v- (not ideal but should work).

couldn't the names start with those prefixes anyway (especially after (un)manage for kvm is merged)? Not sure how feasible it is to get information about that from the MS during this process. Maybe those are guaranteed to have the correct <rate/>,...

**Comment by weizhouapache** [comment]
> > > LibvirtComputingResource
> > 
> > 
> > I think option 2 is better. we could skip the vms if the name does not start with i-/r-/s-/v- (not ideal but should work).
> 
> couldn't the names start with those prefixes anyway (especially after (un)manage for kvm is merged)? Not sure how feasible it is to get information about that from the MS during this process. Maybe those are guaranteed to have the correct ,...

@DaanHoogland 
I think the challenge is, cloudstack agent cannot get the list of available vms from management server during start. there are few options
(1) determine if vm belongs to cloudstack by vm name (prefix), but it will not work if we import unmanaged vms in 4.19 as Daan said.
(2) determine if vm belongs to cloudstack by sysinfo, which were added in commit 3894d34b
```
  <sysinfo type='smbios'>
    <system>
      <entry name='manufacturer'>Apache Software Foundation</entry>
      <entry name='product'>CloudStack KVM Hypervisor</entry>
      <entry name='uuid'>0820a0ca-f7e8-470a-a4f9-7a58fb75a6e4</entry>
    </system>
  </sysinfo>
```

(3) add a try-catch block and ignore the error so that cloudstack-agent can be started even if the vm xml cannot be parsed.

(4) move the `setupMemoryBalloonStatsPeriod` step out from start process (added in #6358).


option 3 looks like the most simple.

**Comment by harikrishna-patnala** [comment]
Yes agreed @weizhouapache I'm working on option 3

**Comment by DaanHoogland** [comment]
should be fixed by #8049
